apps/storybook/src/ComboBox.stories.tsx

```diff
@@ -478,6 +479,81 @@ WithCustomMessageIcon.args = {
   ),
 };
 
+export const Loading: Story<Partial<ComboBoxProps<string>>> = (args) => {
+  const [options, setOptions] = React.useState<SelectOption<string>[]>([]);
+  const [selectedValue, setSelectedValue] = React.useState<string>();
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  const onChange = React.useCallback((value: string) => {
+    action(value ?? '')();
+    setSelectedValue(value);
+  }, []);
+
+  const emptyContent = React.useMemo(() => {
+    return isLoading ? (
+      <>
+        {new Array(6).fill(null).map((_, index) => {
+          // Randomize skeletons width
+          const normalizedIndex = (index % 2) + 1;
+          const randomValue = normalizedIndex * Math.random() * 60;
+          const boundedWidth = Math.min(Math.max(randomValue, 25), 60);
+          return (
+            <MenuItemSkeleton
+              key={index}
+              hasIcon
+              hasSublabel
+              contentWidth={boundedWidth}
+            />
+          );
+        })}
+      </>
+    ) : (
+      'No options found'
+    );
+  }, [isLoading]);
+
+  return (
+    <ComboBox
+      inputProps={{ placeholder: 'Select a country' }}
+      value={selectedValue}
+      onChange={onChange}
+      {...args}
+      emptyStateMessage={emptyContent}
+      options={options}
+      onExpand={React.useCallback(() => {
+        if (!isLoading) {
+          return;
+        }
+        // Mock API request
+        setTimeout(() => {
+          setIsLoading(false);
+          setOptions(
+            countriesList.map(
+              (country) =>
+                ({
+                  ...country,
+                  sublabel: country.value,
+                  icon: (
+                    <img
+                      loading='lazy'
+                      style={{ width: 20, height: 15 }}
+                      src={`https://flagcdn.com/w20/${country.value.toLowerCase()}.png`}
+                      srcSet={`https://flagcdn.com/w40/${country.value.toLowerCase()}.png 2x`}
+                      alt=''
+                    />
+                  ),
+                } as SelectOption<string>),
+            ),
+          );
+        }, 2000);
```

Comment:
maybe move this into a helper so that the logic becomes simpler
```suggestion
        if (isLoading) {
          const _options = await fetchOptions();
          setIsLoading(false);
          setOptions(_options);
        }
```